### PR TITLE
Fix Gutenberg e2e tests failures.

### LINKED ISSUE #2993: investigate why is WP 5.4 + GB E2E test is failing and remove the skip
In https://github.com/woocommerce/woocommerce-gutenberg-products-block/pull/2990, we added an `allow_fail` for WP 5.4 + GB E2E test because it kept failing for us on main, the reasons are not clear yet, but it could be an API change.

### PR DESCRIPTION
**Pull request by budzanowski**
There was a recent change in how the blocks insert mechanism works. It now works as a popover: https://github.com/WordPress/gutenberg/pull/24429

 This means that inserting a block should close the modal and put the focus on the newly added block. Since the automatic close action is part of the new behavior the `insertBlock` puppeteer command was also updated. It now expects that the blocks selection panel becomes closed and focus is shifted:
https://github.com/WordPress/gutenberg/blob/master/packages/e2e-test-utils/src/inserter.js#L135

We have three instances of a test where we test that the block can be added just once on the page. Since the second add operation is not possible the block selector will not be closed and the focus will not be shifted. This means that we can't use `insertBlock` for our tests. A new utility function was created that tries to add the block but does not wait for the closing. 

Fixes #2993 

### How to test the changes in this Pull Request:

1. This fixes E2E tests triggered by Travis so the tests should be green.

### CODE REVIEW

File: tests/e2e/utils.js [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+ */
+import { searchForBlock } from '@wordpress/e2e-test-utils';
+
+/**
+ * Opens the inserter, searches for the given term, then selects the first
+ * result that appears.
+ *
+ * @param {string} searchTerm The text to search the inserter for.
+ */
+export async function insertBlockDontWaitForInsertClose( searchTerm ) {
```
[the remark this answers is not in the document]
Comment (haszari) [reply tests/e2e/utils.js:12]: Update: I ran this locally (with `trunk`) to better understand the issue. Here's the `all products` test fail:

```
 FAIL  tests/e2e/specs/backend/all-products.test.js (33.901s)
  ● All Products Block › can only be inserted once

    : Timeout - Async callback was not invoked within the 30000ms timeout specified by jest.setTimeout.Timeout - Async callback was not invoked within the 30000ms timeout specified by jest.setTimeout.Error:
```

So the issue is that something's timing out in the test, maybe the `insertBlock()`.

**Review by haszari — APPROVED** [review]
Thanks for sorting this out @budzanowski !

This approach seems reasonable. I tested by re-running the tests locally on `trunk` (reproduced the bug) and then again on this branch, and confirmed this fixes the problem. 🎉 

Note – I manually set up gutenberg plugin and latest e2e utils as appropriate, using the commands in the travis config:

- `npm run wp-env run tests-cli "wp plugin install gutenberg --activate"`
- `npm install @wordpress/e2e-test-utils@latest`

I'm approving the PR, but I wonder if it's better to revisit the tests that were failing. Here are a couple of ideas:

- Perhaps we can tweak how we implement `can only be inserted once` tests so they are more robust across WP versions. E.g. perhaps we can search the inserter for the block and confirm that it's hidden or disabled. Ideally we can do this using utils from WordPress core.
- Or perhaps these tests are not actually that useful - maybe they aren't the most important thing to test. 

Up to you if you want to try any of this, not a blocker - feel free to go with your existing solution. 

(By the way, the inline comments on the code are mostly me figuring out the failure mode, and understanding the proposed solution.)

File: tests/e2e/utils.js (hunk 1 above)
[the remark this answers is not in the document]
Comment (budzanowski) [reply tests/e2e/utils.js:12]: Exactly the failure was inside the `insertBlock()` that was waiting for the inserter to close https://github.com/WordPress/gutenberg/blob/master/packages/e2e-test-utils/src/inserter.js#L135 that is why we have our own function that does not wait for this behavior.

**Comment by budzanowski** [comment]
@haszari `Or perhaps these tests are not actually that useful` that is what I generally think so in this case I would just go with this simple patch and do no overthink it too much. Thanks for the review.

**Pull request merged**